packages/dynamite/dynamite/lib/src/builder/resolve_type.dart

```diff
@@ -120,7 +120,7 @@ TypeResult resolveType(
         );
       case openapi.SchemaType.object:
         if (schema.properties == null) {
-          if (schema.additionalProperties == null) {
+          if (schema.parsedAdditionalProperties == openapi.Schema()) {
             result = TypeResultBase(
               'JsonObject',
```

Comment:
This should be a `BuiltMap<String, JsonObject>`. With additional properties we know we have a Map with Strings as keys and an unknown type as values. JsonObject could also be any other class that implements JsonObject which is not correct.